boards/others/esp32c3_supermini/Kconfig.defconfig

```diff
@@ -0,0 +1,9 @@
+# Copyright (c) 2024 Espressif Systems (Shanghai) Co., Ltd.
+# SPDX-License-Identifier: Apache-2.0
+
+config HEAP_MEM_POOL_ADD_SIZE_BOARD
+	int
+	default 65535 if WIFI && BT
```

Comment:
You don't need to specify heap according to used RF part anymore. Please take a look on Espressif boards.

Reply:
Perhaps I'm misunderstanding re: HEAP_MEM_POOL_ADD_SIZE_BOARD
My quick and dirty test suggests that for ``esp32c3`` the default pool sizes are required, or RF devices silently fail to init.

I did notice that the ``esp32s3`` boards don't have these defaults defined.

Test results (no optimization:
WIFI only:
*** Booting Zephyr OS build v3.7.0-3069-g626fc4e7c5aa ***
Hello from esp32c3_supermini/esp32c3
Heap Stats:
  Free bytes: 22940
  Allocated bytes: 28008
  Max allocated bytes: 28036
uart:~$ wifi connect -s "xxxx" -p "yyyyyyyy" -k 1
Connection requested
Connected
[00:00:15.778,000] <inf> net_dhcpv4: Received: 192.168.0.101
Heap Stats:
  Free bytes: 20536
  Allocated bytes: 30352
  Max allocated bytes: 43532


WIFI + BT:
*** Booting Zephyr OS build v3.7.0-3069-g626fc4e7c5aa ***
Hello from esp32c3_supermini/esp32c3
Heap Stats: 
  Free bytes: 37276
  Allocated bytes: 28008
  Max allocated bytes: 28036
Initializing Bluetooth
[00:00:00.033,000] <inf> esp32_bt_adapter: BT controller compile version [963cad4]
[00:00:00.078,000] <inf> bt_hci_core: Identity: D8:3B:DA:2F:66:AE (public)
[00:00:00.078,000] <inf> bt_hci_core: HCI: version 5.0 (0x09) revision 0x0016, manufacturer 0x02e5
[00:00:00.078,000] <inf> bt_hci_core: LMP: version 5.0 (0x09) subver 0x0016
Entering main loop
Heap Stats: 
  Free bytes: 7564
  Allocated bytes: 57564
  Max allocated bytes: 57564
uart:~$ wifi connect -s "gzorp" -p "gzorpian" -k 1
Connection requested
[00:11:54.631,000] <err> esp32_wifi_adapter: memory allocation failed
[00:11:54.631,000] <err> esp32_wifi_adapter: memory allocation failed
[00:11:54.631,000] <err> esp32_wifi_adapter: memory allocation failed
[00:11:54.631,000] <err> esp32_wifi_adapter: memory allocation failed
[00:11:54.631,000] <err> esp32_wifi_adapter: memory allocation failed
Connected
[00:11:59.842,000] <inf> net_dhcpv4: Received: 192.168.0.101
Heap Stats: 
  Free bytes: 5236
  Allocated bytes: 59836
  Max allocated bytes: 64548

Reply:
The Zephyr heap size request (using the `CONFIG_HEAP_MEM_POOL_ADD_*` is working as before, but it is using the statically allocated heap. Due to ESP32** memory layout, some parts couldn't be re-claimed during the runtime, therefore `CONFIG_ESP_RUNTIME_HEAP` was introduced, which can use those memory areas. As a result, we don't need to specify the static heap size addition and use the runtime heap instead. The system heap requirements become standard and can be used for other subsystems/application.

Reply:
Ok, I think I understand what is going on and have gotten this working by setting CONFIG_HEAP_MEM_POOL_SIZE in a sample app's prj.conf (along with config settings to enable BT and WiFi). 

However, setting CONFIG_HEAP_MEM_POOL_SIZE within the boards Kconfig.defconf doesn't work because the build fails with undefined k_malloc(), due (I think) to the expressif wifi blobs having been complied with heap API dependency and something to do with how the linker follows config file order.

Reply:
Please take a look at the Espressif supported board `esp32c3_devkitm`. You don't need to specify more size for the system heap as there is a new heap that the Wifi adapter is utilizing. Also don't forget to rebase your PR to get the new heap in place.

Reply:
Thanks for the pointer (you meant ``esp32c3_devkitc`` (``esp32c3_devkitm`` hasn't been updated yet).

I had been looking for how to remove ``config HEAP_MEM_POOL_ADD_SIZE_BOARD`` altogether - which now obviously was not going to work after looking through heap code.  It was a good exercise to become more familiar with the code.

Meanwhile, your pr #77877 happened and now it all makes sense.